asserts/validation_set.go

```diff
@@ -130,21 +130,21 @@ func checkValidationSetSnap(snap map[string]interface{}) (*ValidationSetSnap, er
 		return nil, fmt.Errorf(`cannot specify revision %s at the same time as stating its presence is invalid`, what)
 	}
 
-	components, err := checkValidationSetComponents(snap, what)
+	components, err := checkValidationSetComponents(snap, snapRevision, snapName, what)
 	if err != nil {
 		return nil, err
 	}
 
 	return &ValidationSetSnap{
-		Name:       name,
+		Name:       snapName,
 		SnapID:     snapID,
 		Presence:   presence,
 		Revision:   snapRevision,
 		Components: components,
 	}, nil
 }
 
-func checkValidationSetComponents(snap map[string]interface{}, what string) (map[string]ValidationSetComponent, error) {
+func checkValidationSetComponents(snap map[string]interface{}, snapRevision int, snapName string, what string) (map[string]ValidationSetComponent, error) {
```

Comment:
unrelated but it's actually unclear that "what" ends up actually used in errors, maybe we can leave it out and pass snapName first to the function?